### PR TITLE
fix(ui): fix the display of the icon

### PR DESCRIPTION
**Pull request by stonebuzz**
Fix the display of the icon

Before : 
image

After : 
image

| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #number

### CODE REVIEW

**Review by trasher — APPROVED** [review]

**Comment by cedric-anne** [comment]
The best solution would be to have an icon in each dropdown option.

**Comment by stonebuzz** [comment]
> The best solution would be to have an icon in each dropdown option.

image

**Review by orthagh — APPROVED** [review]

File: src/Dropdown.php [hunk 1]
```diff
@@ -863,18 +863,43 @@ public static function dropdownIcons($myname, $value, $store_path, $display = tr
...
+                // templates for select2 dropdown
+                $js = <<<JAVASCRIPT
+                $(function() {
+                    formatFormIcon{$rand} = function(icon) {
+                        if (!icon.id) {
+                            return icon.text;
+                        }
+                        var img = '<span><img alt="" src="{$CFG_GLPI['typedoc_icon_dir']}/'+icon.id+'" />';
+                        var label = '<span>'+icon.text+'</span>';
+                        return $(img+'&nbsp'+label);
```
Comment (cedric-anne) [inline src/Dropdown.php:893]:
```suggestion
                        return $(img+'&nbsp;'+label);
```

**Review by cedric-anne — CHANGES_REQUESTED** [review]

File: src/Dropdown.php [hunk 2]
```diff
@@ -863,18 +863,43 @@ public static function dropdownIcons($myname, $value, $store_path, $display = tr
...
                         $options
                     )
                 );
+
+                global $CFG_GLPI;
+
+                // templates for select2 dropdown
+                $js = <<<JAVASCRIPT
+                $(function() {
+                    formatFormIcon{$rand} = function(icon) {
```
Comment (cedric-anne) [inline src/Dropdown.php:887]: Will prevent function to be declared in global scope.
```suggestion
                    const formatFormIcon = function(icon) {
```

File: src/Dropdown.php [hunk 3]
```diff
@@ -863,18 +863,43 @@ public static function dropdownIcons($myname, $value, $store_path, $display = tr
...
+                        if (!icon.id) {
```
Comment (cedric-anne) [inline src/Dropdown.php:888]: `'0'` string is not skipped for now.
```suggestion
                        if (!icon.id || icon.id == '0') {
```

File: src/Dropdown.php [hunk 4]
```diff
@@ -863,18 +863,43 @@ public static function dropdownIcons($myname, $value, $store_path, $display = tr
...
+                            return icon.text;
+                        }
+                        var img = '<span><img alt="" src="{$CFG_GLPI['typedoc_icon_dir']}/'+icon.id+'" />';
+                        var label = '<span>'+icon.text+'</span>';
+                        return $(img+'&nbsp;'+label);
+                    };
+                    $("#dropdown_{$myname}{$rand}").select2({
+                        width: '60%',
+                        templateSelection: formatFormIcon{$rand},
+                        templateResult: formatFormIcon{$rand}
```
Comment (cedric-anne) [inline src/Dropdown.php:898]:
```suggestion
                        templateSelection: formatFormIcon,
                        templateResult: formatFormIcon
```

**Review by cedric-anne — CHANGES_REQUESTED** [review]

**Review by cedric-anne — APPROVED** [review]

**Pull request merged**